tests/test_diffusion.py

```diff
@@ -0,0 +1,188 @@
+        pd = pd * np.exp(-1 / (2 * (1 - np.power(sig_xy, 2))) * (
+                        ((np.power(x - mu_x, 2)) / np.power(sig_x, 2)) +
+                        ((np.power(y - mu_y, 2)) / np.power(sig_y, 2)) -
+                        ((2 * sig_xy * (x - mu_x) * (y - mu_y)) / (sig_x * sig_y))))
+
+        return pd
+    # Define multivariate mean (mu) and covariance matrix (sig) parameters
+    if mu is None:
+        mu = [np.mean(lon), np.mean(lat)]
+    sig = [[0.01, 0], [0, 0.01]]
+    for i, x in enumerate(lon):
+        for j, y in enumerate(lat):
+            K[i, j, :] = MVNorm(x, y, mu, sig)
+
+    # Scale and invert (to make bowl of low diffusivity)
+    K /= np.max(K) - np.min(K)
+    K -= np.min(K)-0.2
+    K = 1/K
+
+    return Field('K', K, lon, lat, depth, time, transpose=True)
+
+
+def CreateDummyUV(xdim=200, ydim=200):
+    depth = np.zeros(1, dtype=np.float32)
+    time = np.arange(0., 100000., 100000/2., dtype=np.float64)
+    lon = np.linspace(0, 0.1, xdim, dtype=np.float32)
+    lat = np.linspace(0, 0.1, ydim, dtype=np.float32)
+
+    U = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+    V = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+    ufield = Field('U', U, lon, lat, depth=depth,
+                   time=time, transpose=True)
+    vfield = Field('V', V, lon, lat, depth=depth,
+                   time=time, transpose=True)
+
+    return {'U': ufield, 'V': vfield}
+
+
+def CreateStartField(lon, lat):
+    time = np.arange(0., 100000, 100000/2., dtype=np.float64)
+    # An evenly distributed starting density
+    data = np.ones((lon.size, lat.size, time.size), dtype=np.float32)
+    return data
+
+
+def LagrangianDiffusionNoCorrection(particle, grid, time, dt):
+    # Version of diffusion equation with no determenistic term i.e. brownian motion
+    to_lat = 1 / 1000. / 1.852 / 60.
+    to_lon = to_lat / math.cos(particle.lat*math.pi/180)
+    r_var = 1/3.
+    Rx = random.uniform(-1., 1.)
+    Ry = random.uniform(-1., 1.)
+    Kfield = grid.K[time, particle.lon, particle.lat]
+    Rx_component = Rx * math.sqrt(2 * Kfield * dt / r_var) * to_lon
+    Ry_component = Ry * math.sqrt(2 * Kfield * dt / r_var) * to_lat
+    particle.lon += Rx_component
+    particle.lat += Ry_component
+
+
+def UpdatePosition(particle, grid, time, dt):
+    particle.prev_lon = particle.new_lon
+    particle.new_lon = particle.lon
+    particle.prev_lat = particle.new_lat
+    particle.new_lat = particle.lat
+
+
+# Recovery Kernal for particles that diffuse outside boundary
+def Send2PreviousPoint(particle):
+    # print("Recovery triggered at %s | %s!" % (particle.lon, particle.lat))
+    # print("Moving particle back to %s | %s" % (particle.prev_lon, particle.prev_lat))
+    particle.lon = particle.prev_lon
+    particle.lat = particle.prev_lat
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def diffusion_test(mode, type='true_diffusion', particles=1000, timesteps=1000, output_file='diffusion_test'):
```

Comment:
Since there are no `asserts` in the function, py.test can't check anything. Also, for py.test to work a function name has to start with `test_`. Run `py.test -v -s tests/test_diffusion.py` to check